code/game/machinery/vending/_vending.dm

```diff
@@ -138,17 +138,17 @@
 /obj/machinery/vending/attackby(obj/item/W, mob/user)
 
 	var/obj/item/card/id/I = W.GetIdCard()
+	var/obj/item/charge_stick/CS = W.GetChargeStick()
```

Comment:
Since logically `GetChargeStick()` on a charge card would return `src` you probably don't need the `istype()` on W below.